packages/flutter/test/rendering/editable_test.dart

```diff
@@ -587,4 +592,64 @@ void main() {
     editable.layout(BoxConstraints.loose(const Size(1000.0, 1000.0)));
     expect(editable.maxScrollExtent, equals(10));
   });
+
+  test('attach and detach correctly handle gesture', () {
+    _GestureBindingSpy();
+    final TextSelectionDelegate delegate = FakeEditableTextState();
+    final RenderEditable editable = RenderEditable(
+      backgroundCursorColor: Colors.grey,
+      selectionColor: Colors.black,
+      textDirection: TextDirection.ltr,
+      cursorColor: Colors.red,
+      offset: ViewportOffset.zero(),
+      textSelectionDelegate: delegate,
+      text: const TextSpan(
+        text: 'test',
+        style: TextStyle(
+          height: 1.0, fontSize: 10.0, fontFamily: 'Ahem',
+        ),
+      ),
+      startHandleLayerLink: LayerLink(),
+      endHandleLayerLink: LayerLink(),
+      selection: const TextSelection(
+        baseOffset: 0,
+        extentOffset: 3,
+        affinity: TextAffinity.upstream,
+      ),
+      onSelectionChanged: (_, __, ___) { },
+    );
+    final PipelineOwner owner = PipelineOwner(onNeedVisualUpdate: () { });
+    editable.attach(owner);
+    // This should register pointer into GestureBinding.instance.pointerRouter.
+    editable.handleEvent(const PointerDownEvent(), BoxHitTestEntry(editable, const Offset(10,10)));
+    // ignore: avoid_as
+    expect((GestureBinding.instance.pointerRouter as _PointerRouterSpy).routeCount, greaterThan(0));
+    editable.detach();
+    // ignore: avoid_as
+    expect((GestureBinding.instance.pointerRouter as _PointerRouterSpy).routeCount, 0);
+  });
+}
+
+class _GestureBindingSpy extends AutomatedTestWidgetsFlutterBinding {
+  final PointerRouter _testPointerRouter = _PointerRouterSpy();
+
+  @override
+  PointerRouter get pointerRouter {
```

Comment:
even better
```dart
PointerRouter get pointerRouter => _testPointerRouter;
```

Reply:
Ho yes :) I put bracket to be able to put a breakpoint on it ^^ forget to put the arrow style later . Thanks